catalog/core/catalog-core-commands/src/test/java/org/codice/ddf/commands/catalog/TestSeedCommand.java

```diff
@@ -0,0 +1,280 @@
+                new boolean[] {false, false, true});
+        final int expectedResourceRequests = 3;
+
+        runCommandAndVerifyResourceRequests(expectedResourceRequests, resourceRequests -> {
+            assertThat(resourceRequests, hasSize(expectedResourceRequests));
+            verifyResourceRequest(resourceRequests.get(0), id1);
+            verifyResourceRequest(resourceRequests.get(1), id2);
+            verifyResourceRequest(resourceRequests.get(2), id1);
+        }, siteNames -> assertThat(siteNames, is(newArrayList(id1, id2, id1))));
+
+        verify(framework, times(2)).query(any(QueryRequest.class));
+    }
+
+    private void runCommandAndVerifyResourceRequests(int expectedResourceRequests,
+            Consumer<List<ResourceRequest>> requestAssertions,
+            Consumer<List<String>> siteNameAssertions) throws Exception {
+        seedCommand.executeWithSubject();
+
+        ArgumentCaptor<ResourceRequest> resourceRequestCaptor = ArgumentCaptor.forClass(
+                ResourceRequest.class);
+        ArgumentCaptor<String> siteNameCaptor = ArgumentCaptor.forClass(String.class);
+        verify(framework,
+                times(expectedResourceRequests)).getResource(resourceRequestCaptor.capture(),
+                siteNameCaptor.capture());
+
+        List<ResourceRequest> resourceRequests = resourceRequestCaptor.getAllValues();
+        requestAssertions.accept(resourceRequests);
+
+        List<String> siteNames = siteNameCaptor.getAllValues();
+        siteNameAssertions.accept(siteNames);
+    }
+
+    private void mockQueryResponse(int stopReturningResultsAtIndex, String[] ids, boolean[] cached)
+            throws Exception {
+        List<Result> results = new ArrayList<>();
+        for (int i = 0; i < ids.length; ++i) {
+            String id = ids[i];
+            Result mockResult = mock(Result.class);
+            MetacardImpl metacard = new MetacardImpl();
+            metacard.setId(id);
+            metacard.setSourceId(id);
+            metacard.setAttribute("internal.local-resource", cached[i]);
+            when(mockResult.getMetacard()).thenReturn(metacard);
+            results.add(mockResult);
+        }
+
+        QueryResponse response = mock(QueryResponse.class);
+        when(response.getResults()).thenReturn(results);
+        doReturn(response).when(framework)
+                .query(argThat(isQueryWithStartIndex(request -> request.getQuery()
+                        .getStartIndex() < stopReturningResultsAtIndex)));
+
+        QueryResponse noResults = mock(QueryResponse.class);
+        when(noResults.getResults()).thenReturn(Collections.emptyList());
+        doReturn(noResults).when(framework)
+                .query(argThat(isQueryWithStartIndex(request -> request.getQuery()
+                        .getStartIndex() >= stopReturningResultsAtIndex)));
+    }
+
+    private IsQueryWithStartIndex isQueryWithStartIndex(Predicate<QueryRequest> test) {
```

Comment:
Will refactor `IsQueryWithStartIndex` to `QueryWithStartIndex` and `isQueryWithStartIndex` to `queryWithStartIndex`. Then the mockito call will look like: `doReturn(noResults).when(framework).query(argThat(is(queryWithStartIndex(...))));`